fips-check.sh

```diff
@@ -9,118 +9,242 @@
 # This should check out all the approved flavors. The command line
 # option selects the flavor. The keep option keeps the output
 # directory.
-#
-# Some variables may be overridden on the command line.
+
+# These variables may be overridden on the command line.
+MAKE="${MAKE:-make}"
+GIT="${GIT:-git -c advice.detachedHead=false}"
+TEST_DIR="${TEST_DIR:-XXX-fips-test}"
+FLAVOR="${FLAVOR:-linux}"
+KEEP="${KEEP:-no}"
+FIPS_REPO="${FIPS_REPO:-git@github.com:wolfssl/fips.git}"
 
 Usage() {
     cat <<usageText
 Usage: $0 [flavor] [keep]
 Flavor is one of:
-    netbsd-selftest
-    marvell-linux-selftest
     linuxv2 (FIPSv2, use for Win10)
-    wolfrand
+    fipsv2-OE-ready (ready FIPSv2)
     solaris
+    netbsd-selftest
+    marvell-linux-selftest
     linuxv5 (current FIPS 140-3)
     fips-ready (ready FIPS 140-3)
     fips-dev (dev FIPS 140-3)
-Keep (default off) retains the XXX-fips-test temp dir for inspection.
+    wolfrand
+Keep (default off) retains the temp dir $TEST_DIR for inspection.
 
 Example:
     $0 windows keep
 usageText
 }
 
-# These variables may be overridden on the command line.
-MAKE="${MAKE:-make}"
-GIT="${GIT:-git -c advice.detachedHead=false}"
-TEST_DIR="${TEST_DIR:-XXX-fips-test}"
-FLAVOR="${FLAVOR:-linux}"
-KEEP="${KEEP:-no}"
-
 while [ "$1" ]; do
   if [ "$1" = 'keep' ]; then KEEP='yes'; else FLAVOR="$1"; fi
   shift
 done
 
-FIPS_REPO='git@github.com:wolfssl/fips.git'
-CRYPT_INC_PATH='wolfssl/wolfcrypt'
-CRYPT_SRC_PATH='wolfcrypt/src'
-
 case "$FLAVOR" in
-linuxv2|fipsv2-OE-ready)
+linuxv2|fipsv2-OE-ready|solaris)
   FIPS_OPTION='v2'
-  FIPS_VERSION='WCv4-stable'
-  CRYPT_VERSION='WCv4-stable'
-  RNG_VERSION='WCv4-rng-stable'
-  WC_MODS=('aes' 'aes_asm' 'cmac' 'des3' 'dh' 'ecc' 'hmac' 'random' 'rsa' 'sha' 'sha256' 'sha3' 'sha512')
-  FIPS_SRCS=('fips.c' 'fips_test.c' 'wolfcrypt_first.c' 'wolfcrypt_last.c')
-  FIPS_INCS=('fips.h')
+  FIPS_FILES=('WCv4-stable'
+    'wolfcrypt/src/fips.c'
```

Comment:
don't you need to do the same things for these that you did for `WOLFCRYPT_FILES`, i.e. be able to separately designate the tag for each file?  there's definitely OE-specific stuff in the fips files themselves, not least the linuxkm stuff.

Reply:
Fair enough. I was trying to short cut this because the fips repo checkout is obnoxious due to all the test vector files. You made a good point about the linuxkm OE.